integration_test/third_party_apps_test.go

```diff
@@ -377,57 +388,107 @@ func TestThirdPartyApps(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	for _, platform := range testConfig.Platforms {
-		platform := platform // https://golang.org/doc/faq#closures_and_goroutines
-		t.Run(platform, func(t *testing.T) {
-			t.Parallel()
+	type test struct {
+		platform   string
+		app        string
+		skipReason string
+	}
+	tests := []test{}
+	platforms := testConfig.Platforms
+	for _, platform := range platforms {
+		apps, err := appsToTest(agentType, platform)
+		if err != nil {
+			t.Fatalf("Error when reading list of apps to test for agentType=%v, platform=%v. err=%v", agentType, platform, err)
+		}
+		if len(apps) == 0 {
+			t.Fatalf("Found no applications when testing agentType=%v, platform=%v", agentType, platform)
+		}
+		for _, app := range apps {
+			tests = append(tests, test{platform, app, ""})
+		}
+	}
 
-			apps, err := appsToTest(agentType, platform)
-			if err != nil {
-				t.Fatalf("Error when reading list of apps to test for agentType=%v, platform=%v. err=%v", agentType, platform, err)
-			}
-			if len(apps) == 0 {
-				t.Fatalf("Found no applications when testing agentType=%v, platform=%v", agentType, platform)
+	// Filter tests
+	mf := modifiedFiles(t)
+	impactedApps := make(map[string]bool)
+	for _, f := range mf {
+		if strings.HasPrefix(f, "apps/") {
+			// File names: apps/<appname>.go
+			f := strings.TrimPrefix(f, "apps/")
+			f = strings.TrimSuffix(f, ".go")
+			impactedApps[f] = true
+
+		} else if strings.HasPrefix(f, "integration_test/third_party_apps_data/applications/") {
+			// Folder names: integration_test/third_party_apps_data/applications/<app_name>
+			f := strings.TrimPrefix(f, "integration_test/third_party_apps_data/applications/")
+			f = strings.Split(f, "/")[0]
+			impactedApps[f] = true
+
+		} else if strings.HasPrefix(f, "integration_test") {
+			// Any other modified file in the directory probably
+			// means we should test everything.
+			impactedApps = make(map[string]bool)
```

Comment:
SGTM.  I thought this was cleaner.

Reply:
it's not about it being clean, there's a correctness issue. resetting it to empty doesn't do the same thing as forcing it to test everything. What if you decide to test everything and thereby set it to empty, then a later modified file makes you decide that you should test apache? Then the map will contain just apache, and you'll end up testing just apache instead of everything.

Reply:
Done.